docs/docker_run.rst

```diff
@@ -0,0 +1,190 @@
+task temporary volume. The file can also be downloaded to a specific ``Volume`` by
+specifying a volume parameter, as follows:
+
+.. code-block:: python
+
+    vol = Volume(host_path, container_path)
+    docker_run.delay('my/image', pull_image=True,
+        container_args=[GirderFileIdToVolume(file_id,volume=vol)])
+
+If the being downloaded is particularly large you may want to consider streaming
+it into the container using a named pipe. See `Streaming Girder files into a container`_
+for more details.
+
+Uploading files to Girder items
+-------------------------------
+
+Utility classes are also provide to simplify uploading files generated by a the
+docker container. The ``GirderUploadVolumePathToItem`` provides the functionality
+to upload a file to an item. In the example below, we using the ``VolumePath``
+utility class to define a file path that we then pass to the docker container.
+The docker container can write data to this file path. As well as passing the
+``VolumePath`` instance as a container argument we also pass it to ``GirderUploadVolumePathToItem``
+, the ``GirderUploadVolumePathToItem`` instance is add to ``girder_result_hooks``. This
+tell girder_work to upload the file path to the item id provided when the docker
+container has finished running.
+
+
+.. code-block:: python
+
+    volumepath = VolumePath('write_data_to_be_upoaded.txt')
+    docker_run.delay('my/image', pull_image=True, container_args=[volumepath],
+        girder_result_hooks=[GirderUploadVolumePathToItem(volumepath, item_id)])
+
+
+Using named pipes to stream data in and out of containers
+---------------------------------------------------------
+
+girder_worker uses named pipes as a language agnostic way of streaming data in
+and out of docker containers. Basically a named pipe is created at a path that is
+mounted into the container. This allows the container to open that pipe for read or
+write and similarly the girder_worker infrastructure can open the pipe on the host,
+thus allow data write and read from the container.
+
+The are two utility classes using to represent as named pipe, ``NamedOutputPipe``
+and ``NamedInputPipe``
+
+NamedOuputPipe
+^^^^^^^^^^^^^^
+
+This represents a named pipe that will be opened in the docker container for write.
+So is using to stream data out of a container.
+
+
+NamedInputPipe
+^^^^^^^^^^^^^^
+
+This represents a named pipe that will be opened in the docker container for read.
+So is using to stream data into a container.
+
+These pipes can be connected to other streams using the ``Connect`` untility class.
+
+
+Streaming Girder files into a container
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One great example of using a named pipe is to stream a potentially large file into
+a container. This approach allow the task to start process straight away rather
+than have to wait for the either file to download, it also removed the requirement
```

Comment:
than having* to wait for the entire* file to download. It also removes* the requirement